### ISSUE TITLE
show method for chains

### ISSUE THREAD

**Issue by mkborregaard** [TuringLang/MCMCChains.jl#28]
`Chains` need a proper `show(IO, Chains)` method.
```julia
julia> chain = sample(f(1), HMC(500,0.01,1))
[HMC] Sampling...100% Time: 0:00:03
[HMC] Finished with
  Running time        = 2.8699147269999976;
  Accept rate         = 1.0;
  #lf / sample        = 0.998;
  #evals / sample     = 2.998;
  pre-cond. metric    = [1.0].
Turing.Chain{AbstractRange{Int64}}(0.0, Turing.Sample[Sample(0.002, Dict{Symbol,Any}(:C=>0.844339,:lf_num=>0,:A=>1.04611,:B=>1.00852,:elapsed=>0.000361018,:epsilon=>0.01,:lp=>-5.58795,:eval_num=>2,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.848448,:lf_num=>1,:A=>1.05759,:B=>1.02014,:elapsed=>2.76864,:epsilon=>0.01,:lp=>-5.61528,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.846487,:lf_num=>1,:A=>1.064,:B=>1.03292,:elapsed=>0.000542769,:epsilon=>0.01,:lp=>-5.63354,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.842883,:lf_num=>1,:A=>1.07216,:B=>1.02505,:elapsed=>0.000310945,:epsilon=>0.01,:lp=>-5.63111,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.828432,:lf_num=>1,:A=>1.07779,:B=>1.03559,:elapsed=>0.00023048,:epsilon=>0.01,:lp=>-5.63594,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.808486,:lf_num=>1,:A=>1.07735,:B=>1.04103,:elapsed=>0.000177207,:epsilon=>0.01,:lp=>-5.62479,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.811408,:lf_num=>1,:A=>1.09045,:B=>1.04669,:elapsed=>0.000159575,:epsilon=>0.01,:lp=>-5.64726,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.814945,:lf_num=>1,:A=>1.09428,:B=>1.04883,:elapsed=>0.000145044,:epsilon=>0.01,:lp=>-5.65657,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.793754,:lf_num=>1,:A=>1.08999,:B=>1.0585,:elapsed=>0.00014354,:epsilon=>0.01,:lp=>-5.64502,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.796201,:lf_num=>1,:A=>1.09549,:B=>1.04152,:elapsed=>0.000123191,:epsilon=>0.01,:lp=>-5.63516,:eval_num=>3,:lf_eps=>0.01))  …  Sample(0.002, Dict{Symbol,Any}(:C=>0.52954,:lf_num=>1,:A=>1.2853,:B=>1.03994,:elapsed=>0.000119147,:epsilon=>0.01,:lp=>-5.6827,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.532742,:lf_num=>1,:A=>1.2792,:B=>1.03433,:elapsed=>0.000119188,:epsilon=>0.01,:lp=>-5.67075,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.546827,:lf_num=>1,:A=>1.27016,:B=>1.04616,:elapsed=>0.00011863,:epsilon=>0.01,:lp=>-5.67914,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.54944,:lf_num=>1,:A=>1.26545,:B=>1.04865,:elapsed=>0.000119486,:epsilon=>0.01,:lp=>-5.67721,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.544144,:lf_num=>1,:A=>1.2696,:B=>1.04755,:elapsed=>0.000119432,:epsilon=>0.01,:lp=>-5.67843,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.539053,:lf_num=>1,:A=>1.26228,:B=>1.03966,:elapsed=>0.000119662,:epsilon=>0.01,:lp=>-5.65816,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.536991,:lf_num=>1,:A=>1.26476,:B=>1.02714,:elapsed=>0.000119664,:epsilon=>0.01,:lp=>-5.64725,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.529395,:lf_num=>1,:A=>1.26652,:B=>1.02576,:elapsed=>0.000118504,:epsilon=>0.01,:lp=>-5.64401,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.534894,:lf_num=>1,:A=>1.26016,:B=>1.03599,:elapsed=>0.000118747,:epsilon=>0.01,:lp=>-5.64945,:eval_num=>3,:lf_eps=>0.01)), Sample(0.002, Dict{Symbol,Any}(:C=>0.53443,:lf_num=>1,:A=>1.25372,:B=>1.04135,:elapsed=>0.000119365,:epsilon=>0.01,:lp=>-5.64667,:eval_num=>3,:lf_eps=>0.01))], [0.844339 0.0 … 2.0 0.01; 0.848448 1.0 … 3.0 0.01; … ; 0.534894 1.0 … 3.0 0.01; 0.53443 1.0 … 3.0 0.01], 1:1:500, ["C", "lf_num", "A", "B", "elapsed", "epsilon", "lp", "eval_num", "lf_eps"], [1], Dict{Symbol,Any}())
```
is hilarious :-D

**Comment by cpfiffer** [comment]
I agree! I'm planning on rooting around in how `ChainSummary` structs are printed, and I'll toss that on the pile.

**Comment by trappmartin** [comment]
#43

**Comment by cpfiffer** [comment]
Closed by #44.
